ziti/src/main/kotlin/org/openziti/ZitiException.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2018-2021 NetFoundry, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openziti
+
+
+
+class ZitiException internal constructor(internal val code: Errors, cause: Throwable? = null)
+    : Exception(code.toString(), cause) {
+
+    internal sealed class Errors {
```

Comment:
how does this translate to java? this seems like a kotlin-ism that won't translate well?

Reply:
this is `internal` not visible to app. this is just to map codes from controller to exception messages